env0/resource_project.go

```diff
@@ -106,8 +114,45 @@ func resourceProjectDelete(ctx context.Context, d *schema.ResourceData, meta int
 	apiClient := meta.(client.ApiClientInterface)
 
 	id := d.Id()
-	err := apiClient.ProjectDelete(id)
-	if err != nil {
+
+	forceDestroy := d.Get("force_destroy").(bool)
+
+	if !forceDestroy {
```

Comment:
split this logic into smaller functions
so it will be more readable
something like
* hasActiveEnv
* (for the big loop) validateNoActivceEnvironments
just a suggestion since its not clean

Reply:
Hi @GiliFaroEnv0 ,

Thank you for the comments.
I'm still going over your comments and will refactor it.
However, before I proceed - I have added this comment in Slack:
https://env0.slack.com/archives/C01QBBEGBEH/p1647378021284789?thread_ts=1647344145.698209&cid=C01QBBEGBEH

Happy to discuss further. Thanks!

Reply:
@GiliFaroEnv0 - applied changes as discussed in the slack thread.
https://github.com/env0/terraform-provider-env0/pull/263/commits/f850c770528f357af8ad4654d4dbf151970ccaa7